.github/workflows/backport-prs.yml

```diff
@@ -0,0 +1,169 @@
+            core.info(`Searching for 'internal/main' issue linked to PR #${pr.number}`);
+            try {
+              const { data: searchResults } = await github.request('GET /search/issues', {
+                q: `is:issue state:open label:"internal/main" repo:${owner}/${repo} in:body #${pr.number}`,
+                advanced_search: true,
+                headers: {
+                  'X-GitHub-Api-Version': '2022-11-28'
+                }
+              });
+            } catch (error) {
+              core.setFailed(`Failed to search for main issue for PR #${pr.number}: ${error.message}`);
+            }
+            if (searchResults.total_count === 0) {
+              core.info(`No 'internal/main' issue found for PR #${pr.number}. Exiting.`);
+              return;
+            }
+            const mainIssue = searchResults.items[0];
+            core.info(`Found main issue: #${mainIssue.number}`);
+
+            // https://docs.github.com/en/rest/issues/sub-issues?apiVersion=2022-11-28#add-sub-issue
+            core.info(`Fetching sub-issues for main issue #${mainIssue.number}`);
+            try {
+              const { data: subIssues } = await github.request('GET /repos/{owner}/{repo}/issues/{issue_number}/sub_issues', {
+                owner: owner,
+                repo: repo,
+                issue_number: mainIssue.number,
+                headers: {
+                  'X-GitHub-Api-Version': '2022-11-28'
+                }
+              });
+            } catch (error) {
+              core.setFailed(`Failed to fetch sub-issues for issue #${mainIssue.number}: ${error.message}`);
+            }
+            if (subIssues.length === 0) {
+              core.info(`No sub-issues found for issue #${mainIssue.number}. Exiting.`);
+              return;
+            }
+            core.info(`Found ${subIssues.length} sub-issues.`);
+
+            for (const subIssue of subIssues) {
+              const subIssueNumber = subIssue.number;
+              // Find the release label directly on the sub-issue object
+              const releaseLabel = subIssue.labels.find(label => label.name.startsWith('release/v'));
+              if (!releaseLabel) {
+                core.warning(`Sub-issue #${subIssueNumber} has no 'release/v...' label. Skipping.`);
+                continue;
+              }
+              const targetBranch = releaseLabel.name
+              core.info(`Processing sub-issue #${subIssueNumber} for target branch: ${targetBranch}`);
+              const newBranchName = `backport-${pr.number}-${targetBranch.replace(/\//g, '-')}`;
+              try {
+                execSync(`git config user.name "github-actions[bot]"`);
+                execSync(`git config user.email "github-actions[bot]@users.noreply.github.com"`);
+                execSync(`git fetch origin ${targetBranch}`);
+                execSync(`git checkout -b ${newBranchName} origin/${targetBranch}`);
```

Comment:
Sorry if I'm missing this somewhere, but is there a process to clean up these branches when the PR gets merged? Or does that just happen automatically (i seem to recall that being a default setting somewhere in github, but I could be wrong)

Reply:
yes, it gets automatically cleaned up by GitHub